src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calc = {
+    result: 0,
+
+    add(number) {
+      return calc.result + number;
+    },
+
+    subtract(number) {
+      return calc.result - number;
+    },
+
+    multiply(number) {
+      return calc.result * number;
+    },
+
+    divide(number) {
+      return calc.result / number;
+    },
+
```

Comment:
Just for the fun of it

Reply:
Honestly, i don't even understand why 'this.result' doesn't work ... I've spent hour figuring out why the tests didn't pass

Reply:
Lemme explain) 

`this` = current execution context. In other words, in most cases, the value of this is determined by how a function is **called**, and usually, it's an object before the dot)

When we're calling `calculator.operate()`, operate function kind of knows who's **calling** so this, in that context, refers to `Object{} calculator`, but as an argument we're passing just a reference to a `Function{} calculator.add` _(not to be confused with calling a function, in which case `Function{} add `would know who `this` is)_

So, when we're calling operation(callback) function inside `Function{} operate` like that

```
    operate(operation, num) {
      operation(num)

      return this;
    },
```

And `operation = calculator.add`, it doesn't have `this ` bind into it, so one way to let it know who `this `is, is to use the `call() method `_(it calls a function with a given `this ` value and arguments provided individually)._ 

`operation.call(this, num);`

It will take the `this `from the **called** `Function{} operate`

Reply:
Illya, If you see it, I'm sorry if I missed something or wrote something wrong👀